File: 2301-2400/2381-2390/2390M_Removing_Stars_From_a_String/cs/string.cs
// 2390. Removing Stars From a String
// https://leetcode.com/problems/removing-stars-from-a-string/
// T.C.: O(n)
// S.C.: O(n)
public class Solution
{
  public string RemoveStars(string s)
  {
    var sb = new StringBuilder();
    for (var i = 0; i < s.Length; i++)
    {
      if (s[i] == '*')
      {
        if (sb.Length > 0) sb.Length--;
      }
      else
      {
        sb.Append(s[i]);
      }
    }

    return sb.ToString();
  }
}

var s = "leet**cod*e";
var expected = "lecoe";
var result = new Solution().RemoveStars(s);
Console.WriteLine($"{result}, {result == expected}");

s = "erase*****";
expected = "";
result = new Solution().RemoveStars(s);
Console.WriteLine($"{result}, {result == expected}");

s = "abb*cdfg*****x*";
expected = "a";
result = new Solution().RemoveStars(s);
Console.WriteLine($"{result}, {result == expected}");

File: 1501-1600/1551-1560/1556E_Thousand_Separator/string.js
// 1556. Thousand Separator
// https://leetcode.com/problems/thousand-separator/
// T.C.: O(log n)
// S.C.: O(1)
/**
 * @param {number} n
 * @return {string}
 */
var thousandSeparator = function (n) {
  let s = n.toString();
  let result = '';
  for (let i = s.length - 1, count = 0; i >= 0; i--) {
    if (count === 3) {
      result = '.' + result;
      count = 0;
    }
    result = s[i] + result;
    count++;
  }
  return result;
};

var n = 987;
var expected = '987';
var result = thousandSeparator(n);
console.log(result, result === expected);

var n = 1234;
var expected = '1.234';
var result = thousandSeparator(n);
console.log(result, result === expected);

var n = 0;
var expected = '0';
var result = thousandSeparator(n);
console.log(result, result === expected);

var n = 1;
var expected = '1';
var result = thousandSeparator(n);
console.log(result, result === expected);

var n = 123456789;
var expected = '123.456.789';
var result = thousandSeparator(n);
console.log(result, result === expected);
